datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -404,67 +410,136 @@ impl<'a> StatisticsConverter<'a> {
+    /// # use parquet::file::metadata::ParquetMetaData;
+    /// # use datafusion::datasource::physical_plan::parquet::StatisticsConverter;
+    /// # fn get_parquet_metadata() -> ParquetMetaData { unimplemented!() }
+    /// # fn get_arrow_schema() -> Schema { unimplemented!() }
+    /// // Given the metadata for a parquet file and the arrow schema
+    /// let metadata: ParquetMetaData = get_parquet_metadata();
+    /// let arrow_schema: Schema = get_arrow_schema();
+    /// let parquet_schema = metadata.file_metadata().schema_descr();
+    /// // create a converter
+    /// let converter = StatisticsConverter::try_new("foo", &arrow_schema, parquet_schema)
+    ///   .unwrap();
+    /// // get the minimum value for the column "foo" in the parquet file
+    /// let min_values: ArrayRef = converter
+    ///   .row_group_mins(metadata.row_groups().iter())
+    ///  .unwrap();
+    /// ```
+    pub fn row_group_mins<I>(&self, metadatas: I) -> Result<ArrayRef>
+    where
+        I: IntoIterator<Item = &'a RowGroupMetaData>,
+    {
         let data_type = self.arrow_field.data_type();
-        let num_row_groups = metadata.row_groups().len();
 
-        let parquet_schema = metadata.file_metadata().schema_descr();
-        let row_groups = metadata.row_groups();
+        let Some(parquet_index) = self.parquet_index else {
+            return Ok(self.make_null_array(data_type, metadatas));
+        };
 
-        // find the column in the parquet schema, if not, return a null array
-        let Some((parquet_idx, matched_field)) =
-            parquet_column(parquet_schema, self.arrow_schema, self.column_name)
-        else {
-            // column was in the arrow schema but not in the parquet schema, so return a null array
-            return Ok(new_null_array(data_type, num_row_groups));
+        let iter = metadatas
+            .into_iter()
+            .map(|x| x.column(parquet_index).statistics());
+        min_statistics(data_type, iter)
+    }
+
+    /// Extract the maximum values from row group statistics in [`RowGroupMetaData`]
+    ///
+    /// See docs on [`Self::row_group_mins`] for details
+    pub fn row_group_maxes<I>(&self, metadatas: I) -> Result<ArrayRef>
+    where
+        I: IntoIterator<Item = &'a RowGroupMetaData>,
+    {
+        let data_type = self.arrow_field.data_type();
+
+        let Some(parquet_index) = self.parquet_index else {
+            return Ok(self.make_null_array(data_type, metadatas));
         };
 
-        // sanity check that matching field matches the arrow field
-        if matched_field.as_ref() != self.arrow_field {
-            return internal_err!(
-                "Matched column '{:?}' does not match original matched column '{:?}'",
-                matched_field,
-                self.arrow_field
-            );
-        }
+        let iter = metadatas
+            .into_iter()
+            .map(|x| x.column(parquet_index).statistics());
+        max_statistics(data_type, iter)
```

Comment:
The `min_statistcs` and `max_statistics` changes in another PR could still be used here...

Reply:
Yes, indeed -- you are exactly correct. I purposely didn't change min_statistics and max_statistics as I knew you were working on them already